src/script/interpreter.cpp

```diff
@@ -1475,24 +1475,22 @@ static bool HandleMissingData(MissingDataBehavior mdb)
 }
 
 template<typename T>
-bool SignatureHashSchnorr(uint256& hash_out, ScriptExecutionData& execdata, const T& tx_to, uint32_t in_pos, uint8_t hash_type, SigVersion sigversion, const PrecomputedTransactionData& cache, MissingDataBehavior mdb)
+bool SignatureHashSchnorr(uint256& hash_out, ScriptExecutionData& execdata, const T& tx_to, uint32_t in_pos, uint8_t hash_type, V1SigVersion sigversion, const PrecomputedTransactionData& cache, MissingDataBehavior mdb)
 {
     uint8_t ext_flag, key_version;
     switch (sigversion) {
-    case SigVersion::TAPROOT:
+    case V1SigVersion::TAPROOT:
         ext_flag = 0;
         // key_version is not used and left uninitialized.
         break;
-    case SigVersion::TAPSCRIPT:
+    case V1SigVersion::TAPSCRIPT:
         ext_flag = 1;
         // key_version must be 0 for now, representing the current version of
         // 32-byte public keys in the tapscript signature opcode execution.
         // An upgradable public key version (with a size not 32-byte) may
         // request a different key_version with a new sigversion.
         key_version = 0;
         break;
-    default:
-        assert(false);
```

Comment:
looks like you are still changing the code instead of suppressing the incorrect warning?

Reply:
Dropped all code changes from here for now, to look at the current state of things.